pkgs/build-support/gradle/update-deps.nix

```diff
@@ -0,0 +1,154 @@
+{ lib
+, pkgs
+, stdenv
+, writeScript
+, writeShellScript
+, nix
+, gradle
+, openssl
+, mitm-cache
+, jdk
+, nsjail
+, path
+, python3
+}:
+
+# path to the derivation attr (usually the package name)
+{ attrPath ? pname
+# pname (attrPath synonym)
+, pname ? ""
+# the Nix code to get the derivation (this will be put in Bash double quotes)
+# this is relevant for downstream (non-nixpkgs) users
+, code ? "with import ./. { }; ($attrPath)"
+# unlike gradleFlags in normal packages, these flags are escaped so they are allowed to contain spaces
+# also, gradleFlags from configurePhase are used by default
+, gradleFlags ? []
+# attribute overrides for the "source" derivation
+, attrOverrides ? {}
+# run before fetching the dependencies (in sandbox)
+, preBuild ? ""
+# run before starting the sandbox for fetching dependencies
+, preBuildNoSandbox ? ""
+# the gradle commands for fetching the necessary dependencies
+, gradleCommands ? "gradle nixDownloadDeps"
+# run after fetching the dependencies (in sandbox)
+, postBuild ? ""
+# run after the sandbox for fetching dependencies has stopped
+, postBuildNoSandbox ? ""
+, mitmCachePort ? 1337
+# deps path (relative to the package directory), or absolute
+, depsPath ? "deps.json"
+# extra packages to make available in the sandbox
+, availablePackages ? [ ]
+}:
+
+let
+  keep = [ "MITM_CACHE_PORT" "MITM_CACHE_KEYSTORE" "MITM_CACHE_KS_PWD" "MITM_CACHE_CA" ];
+  availablePackages' = map (drv: ''(builtins.storePath "${drv}")'') availablePackages;
```

Comment:
I am not sure if we can use builtins.storePath at all because it is not pure https://github.com/NixOS/nix/blob/419a64f9e04a0e1ff63e12b9581bb4c01b41e822/src/libexpr/primops.cc#L1483

Reply:
is there a better way of pinning dependencies to the ones the user provided? I think the only way would be to completely sidestep nix-shell?

Reply:
fixed